Code/Editor/Platform/Common/Clang/editor_lib_clang.cmake

```diff
@@ -5,5 +5,8 @@
 # SPDX-License-Identifier: Apache-2.0 OR MIT
 #
 #
-
-set(LY_COMPILE_OPTIONS PRIVATE -fexceptions)
+if(MSVC)
+    set(LY_COMPILE_OPTIONS PRIVATE /EHsc)
+else()
+    set(LY_COMPILE_OPTIONS PRIVATE -fexceptions)
+endif()
```

Comment:
Maybe I'm missing a particular critical piece of information here, but why would a file called 'something_something_clang' need to care about MSVC options at all?

Reply:
Because `clang-cl.exe` only supports the MSVC version whle `clang.exe` and `clang++.exe` supports the clang version of options.

Reply:
Microsoft didn't make a custom version of clang.
The actual clang that comes installed with LLVM comes with `clang.exe`, `clang++.exe` and `clang-cl.exe`
![image](https://user-images.githubusercontent.com/56135373/224101437-e1a5139d-a480-4f58-b7f5-dde4f7285823.png)

Reply:
I guess what I mean here is that, our PAL files like "blahblah_compiler.cmake" are largely about the format of flags to feed the compiler, not about the platform.   Which means that if we have a compiler that takes MSVC arguments, we should probably shunt it into blahblah_MSVC.cmake regardless of whether its a clang, Green Hills, Intel, or msvc++ compiler, right?  And if it takes clang-like arguments it should go into blahblah_clang.cmake?

Or is that not appropriate here.  Making sure we don't have an invalid assumption to begin with.

Reply:
Well it is tricky here because the `CMAKE_CXX_COMPILER_ID` is set to "Clang", but the `MSVC` variable is also set.

That being said, I don't there is a need to support `clang-cl` while `clang++` works just as well.

Reply:
Visual Studio may be not the best IDE on Windows, let me say why not.
There's no reason to refuse developers who like VS and clang.

Reply:
> Visual Studio may be not the best IDE on Windows, let me say why not. There's no reason to refuse developers who like VS and clang.

Visual studio can also use clang.exe and clang++.exe, it doesn't have to be clang-cl: https://learn.microsoft.com/en-us/cpp/build/clang-support-cmake?view=msvc-170#clang-compatibility-modes

Reply:
Would the updating the first if statement in Configurations_windows.cmake alliviate the need to update all the msvc files with clang logic?

If we change the if check from
```cmake
if(CMAKE_CXX_COMPILER_ID STREQUAL "MSVC")
```

to

```cmake
if(CMAKE_CXX_COMPILER_ID STREQUAL "MSVC" OR MSVC)
```

Would the other changes be necessary?

Reply:
`clang-cl` does not support some options such as:
`/Zc:lambda`
`/favor:AMD64`

When building with `clang-cl`, the compiler complains some warnings that are disabled by the following options.
```
        -Wno-reorder-ctor
        -Wno-logical-not-parentheses
        -Wno-logical-op-parentheses
        -Wno-switch
        -Wno-undefined-var-template
        -Wno-inconsistent-missing-override
        -Wno-parentheses
        -Wno-unused-parameter
        -Wno-sign-compare
        -Wno-ignored-qualifiers
        -Wno-missing-field-initializers

        # disable warning introduced by -fsized-deallocation, pybind11 used.
        -Wno-unknown-argument
```
`cl` does not support these options.

Reply:
> > > @Jackie9527 As an alternative the `clang++.exe` and `clang.exe` can be used instead of clang-cl on Windows. It is also exposed with the ninja generator via the `windows-ninja-clang` [CMakePreset](https://github.com/o3de/o3de/blob/development/cmake/Platform/Windows/CMakePresets.json#L123-L141).
> > > If anything a Preset should be added for Windows + Visual Studio + Clang Cl
> > 
> > 
> > OK, I'll work on that.
> 
> FYI CMake Presets support supplying the `-T toolset` parameter via the `toolset` field as part of a [configurePreset](https://cmake.org/cmake/help/latest/manual/cmake-presets.7.html#configure-preset)
> 
> > architecture, toolset
> > Optional fields representing the platform and toolset, respectively, for [generators](https://cmake.org/cmake/help/latest/manual/cmake-generators.7.html#manual:cmake-generators(7)) that support them.
> > See [cmake -A](https://cmake.org/cmake/help/latest/manual/cmake.1.html#cmdoption-cmake-A) option for possible values for architecture and [cmake -T](https://cmake.org/cmake/help/latest/manual/cmake.1.html#cmdoption-cmake-T) for toolset.

done

Reply:
I guess it could be difficult for others to read the options, and doesn't match single responsibility protocol

Reply:
I guess it could be difficult for others to read the options, and doesn't match single responsibility protocol.